variation/validators/duplication_deletion_base.py

```diff
@@ -0,0 +1,254 @@
+from variation.validators.validator import Validator
+from variation.hgvs_dup_del_mode import HGVSDupDelMode
+from variation.data_sources import SeqRepoAccess, TranscriptMappings, UTA
+from variation.tokenizers import GeneSymbol
+from variation.mane_transcript import MANETranscript
+from ga4gh.vrs.dataproxy import SeqRepoDataProxy
+from ga4gh.vrs.extras.translator import Translator
+from gene.query import QueryHandler as GeneQueryHandler
+import logging
+from variation.schemas.normalize_response_schema\
+    import HGVSDupDelMode as HGVSDupDelModeEnum
+
+logger = logging.getLogger('variation')
+logger.setLevel(logging.DEBUG)
+
+
+class DuplicationDeletionBase(Validator):
+    """The Deletion Validator Base class."""
+
+    def __init__(self, seq_repo_access: SeqRepoAccess,
+                 transcript_mappings: TranscriptMappings,
+                 gene_symbol: GeneSymbol,
+                 mane_transcript: MANETranscript,
+                 uta: UTA, dp: SeqRepoDataProxy, tlr: Translator,
+                 gene_normalizer: GeneQueryHandler):
+        """Initialize the Deletion Base validator.
+
+        :param SeqRepoAccess seq_repo_access: Access to SeqRepo data
+        :param TranscriptMappings transcript_mappings: Access to transcript
+            mappings
+        :param GeneSymbol gene_symbol: Gene symbol tokenizer
+        :param MANETranscript mane_transcript: Access MANE Transcript
+            information
+        :param UTA uta: Access to UTA queries
+        :param GeneQueryHandler gene_normalizer: Access to gene-normalizer
+        """
+        super().__init__(
+            seq_repo_access, transcript_mappings, gene_symbol, mane_transcript,
+            uta, dp, tlr, gene_normalizer
+        )
+        self.hgvs_dup_del_mode = HGVSDupDelMode(seq_repo_access)
+
+    def get_reference_sequence(
+            self, ac: str, start: int, end: int, errors: List,
+            cds_start: int = None) -> Optional[str]:
+        """Get deleted reference sequence.
+
+        :param str ac: Accession
+        :param int start: Start position
+        :param int end: End position
+        :param list errors: List of errors
+        :param int cds_start: Coding start site
+        :return: Reference sequence of nucleotides
+        """
+        if cds_start:
+            start += cds_start
+            if end is not None:
+                end += cds_start
+
+        if start and not end:
+            ref_sequence = self.seqrepo_access.get_sequence(
+                ac, start
+            )
+        elif start is not None and end is not None:
+            ref_sequence = self.seqrepo_access.get_sequence(
+                ac, start, end
+            )
+        else:
+            ref_sequence = None
+
+        if not ref_sequence:
+            errors.append("Unable to get reference sequence.")
+        return ref_sequence
+
+    def check_reference_sequence(self, t, s, errors, cds_start=None) -> bool:
```

Comment:
types

Reply:
Lol I just realized my last commit specifically updated the docs in this class, but somehow managed to miss this one